### ISSUE TITLE
LSM2 SDPA can produce NaNs for fully masked attention rows

### ISSUE THREAD

**Issue by NarayanSchuetz** [AshleyLab/OpenMHC#11]
## Problem

LSM2 builds additive attention masks for PyTorch SDPA with `-inf` for masked keys. If a retained-token row has no valid keys, the SDPA mask for that row is all `-inf`. Softmax over an all-`-inf` row is undefined and can produce NaN activations.

This is reachable in training when inherited missingness plus artificial masking leaves all kept tokens masked for a sample. The inference path may hide this because outputs are later filtered/replaced, but the internal encoder activations can still become invalid.

## Implications

Training can produce NaN activations and NaN gradients. Inference may appear stable while laundering NaNs downstream, which makes the failure mode harder to notice. Benchmark outputs are mostly expected to be unchanged for normal cases, but extreme missingness cases may change from NaN-contaminated or fallback behavior to finite model behavior after a fix.

## Proposed Fix

Before calling `torch.nn.functional.scaled_dot_product_attention`, detect rows whose additive mask is entirely `-inf` and replace those row masks with zeros so SDPA returns finite values. Rows with at least one valid key should remain unchanged.

Suggested shape-preserving repair in `MultiHeadAttention.forward` after mask broadcasting/dtype conversion:

```python
fully_masked_rows = torch.isneginf(sdpa_mask).all(dim=-1, keepdim=True)
sdpa_mask = sdpa_mask.masked_fill(fully_masked_rows, 0.0)
```

Add a regression test that passes an all-`-inf` attention mask through `MultiHeadAttention`, runs backward, and asserts finite outputs and gradients.
